pkg/transport/generic_tranpsort_controller_test.go

```diff
@@ -151,62 +151,50 @@ type bindingCase struct {
 	ExpectedKeys []any // JSON equivalent of keys of expect, for logging
 }
 
-func clusterScopeKey(elts ksapi.ClusterScopeDownsyncObjects) metav1.GroupVersionResource {
-	return elts.GroupVersionResource
-}
-
-func newClusterScope(gvr metav1.GroupVersionResource) ksapi.ClusterScopeDownsyncObjects {
-	return ksapi.ClusterScopeDownsyncObjects{GroupVersionResource: gvr}
-}
-
-func namespaceScopeKey(elts ksapi.NamespaceScopeDownsyncObjects) metav1.GroupVersionResource {
-	return elts.GroupVersionResource
-}
-
-func newNamespaceScope(gvr metav1.GroupVersionResource) ksapi.NamespaceScopeDownsyncObjects {
-	return ksapi.NamespaceScopeDownsyncObjects{GroupVersionResource: gvr}
-}
-
-func namespaceAndNamesKey(elts ksapi.NamespaceAndNames) string /*namespace*/ {
-	return elts.Namespace
+func newClusterScope(gvr metav1.GroupVersionResource, name, resourceVersion string) ksapi.ClusterScopeDownsyncObject {
+	return ksapi.ClusterScopeDownsyncObject{
+		GroupVersionResource: gvr,
+		Name:                 name,
+		ResourceVersion:      resourceVersion,
+	}
 }
 
-func newNamespaceAndNames(nsName string) ksapi.NamespaceAndNames {
-	return ksapi.NamespaceAndNames{Namespace: nsName}
+func newNamespaceScope(gvr metav1.GroupVersionResource, namespace, name, resourceVersion string) ksapi.NamespaceScopeDownsyncObject {
+	return ksapi.NamespaceScopeDownsyncObject{
+		GroupVersionResource: gvr,
+		Namespace:            namespace,
+		Name:                 name,
+		ResourceVersion:      resourceVersion,
+	}
 }
 
 func (bc *bindingCase) Add(obj mrObjRsc) {
 	key := util.RefToRuntimeObj(obj.MRObject)
-	gvr := metav1.GroupVersionResource{Group: key.GK.Group, Version: obj.MRObject.GetObjectKind().GroupVersionKind().Version, Resource: obj.Resource}
+	gvr := metav1.GroupVersionResource{
+		Group:    key.GK.Group,
+		Version:  obj.MRObject.GetObjectKind().GroupVersionKind().Version,
+		Resource: obj.Resource,
+	}
 	objNS := obj.MRObject.GetNamespace()
 	objName := obj.MRObject.GetName()
+	objRV := obj.MRObject.GetResourceVersion()
```

Comment:
Doesn't this PR also need to update `func (gen *generator) generateObjectMeta`, to make it generate a non-"" ResourceVersion? If I recall correctly, the fakes do nothing with ResourceVersion.